doc/source/data/monitoring-your-workload.rst

```diff
@@ -107,3 +107,92 @@ When an operator completes, the metrics for that operator are also logged.
+~~~~~~~~~~~~~~
+For each operator, a summary on the individual operator's execution stats will be included. This operator will operator
+on many different blocks, so some stats will show the min, max, mean, and sum of the stat aggregated over all the blocks. 
+The following are descriptions of the various stats included at the operator level:
+
+* **Remote wall time**: The wall time is the start to finish time for an operator. It might include time where the operator
+  is not processing data, sleeping, waiting for I/O, etc.
+* **Remote cpu time**: The cpu time is the process time for an operator which will exclude time slept. This will include 
+  both user and system CPU time.
+* **UDF time**: The UDF time is time spent in functions defined by the user. This will include functions passed into Ray 
+  Data methods including :meth:`~ray.data.Dataset.map`, :meth:`~ray.data.Dataset.map_batches`, :meth:`~ray.data.Dataset.filter`,
+  etc. You can use this to track how much time is spent in functions you define and how much time could be gained by optimizing 
+  those functions.
+* **Memory usage**: The memory usage per block is shown in MiB. 
+* **Output stats**: The output includes stats on the number of rows output and size of output in bytes per block. The number of
+  output rows per task are also included. All of this together will give you insight into how much data is being output at a per
+  block and per task level.
+* **Task Stats**: The scheduling of tasks to nodes is shown which will allow you to see if you are utilizing all of your nodes
+  as expected.
+* **Throughput**: Throughput for the operator is calculated and for a point of comparison an estimate of the throughput of the 
+  same task on a single node is computed. This estimate assumes the total time of the work remains the same, but there is no 
+  concurrency. Throughput is also calculated at the dataset level including a single node estimate.
+
+Iterator stats
+~~~~~~~~~~~~~~
+If the data is iterated over, iteration stats are also generated. Even if you are not directly iterating over the data, there might be iteration stats included e.g. if you 
+call :meth:`~ray.data.Dataset.take_all`. Some of the stats included at the iterator level are:
+
+* **Iterator initialization**: The time spent initializing the iterator to be used. This is internal to Ray Data.
+* **Time user thread is blocked**: The time spent producing data in the iterator. Often this will be the primary execution of a
+  dataset if it has not previously been materialized. 
+* **Time in user thread**: The time spent in the user thread that is iterating over the dataset outside of the Ray Data code.
+  If this time is high consider optimizing the body of the loop that is iterating over the dataset.
+* **Batch iteration stats**: Stats are also included about the prefetching of batches. These times should be internal to Ray
+  Data code, but by tuning how prefetching is performed this might be further optimized.
```

Comment:
```suggestion
* **Batch iteration stats**: Stats are also included about the prefetching of batches. These times are internal to Ray
  Data code, but by tuning how prefetching is performed, this can be further optimized.
```